build-llvm.sh

```diff
@@ -134,6 +134,19 @@ if [ -n "$HOST" ]; then
     CMAKEFLAGS="$CMAKEFLAGS -DCLANG_DEFAULT_CXX_STDLIB=libc++"
     CMAKEFLAGS="$CMAKEFLAGS -DCLANG_DEFAULT_LINKER=lld"
     BUILDDIR=$BUILDDIR-$HOST
+    
+    if [ -n "$WITH_PYTHON" ]; then
+        PYTHON_VER="3.8.7"
+        PYTHON_HOME="$PREFIX"
+
+        CMAKEFLAGS="$CMAKEFLAGS -DLLDB_ENABLE_PYTHON=ON"
+        CMAKEFLAGS="$CMAKEFLAGS -DPYTHON_HOME=$PYTHON_HOME"
+        CMAKEFLAGS="$CMAKEFLAGS -DPYTHON_LIBRARIES=$PYTHON_HOME/lib/python3.8/config-3.8/libpython3.8.dll.a"
+        CMAKEFLAGS="$CMAKEFLAGS -DPYTHON_INCLUDE_DIRS=$PYTHON_HOME/include/python3.8"
+        CMAKEFLAGS="$CMAKEFLAGS -DPYTHON_EXECUTABLE=python3"
+        CMAKEFLAGS="$CMAKEFLAGS -DLLDB_PYTHON_RELATIVE_PATH=python-$PYTHON_VER"
+        CMAKEFLAGS="$CMAKEFLAGS -DLLDB_PYTHON_HOME=python-$PYTHON_VER;"
```

Comment:
Is this correct? the LLDB CMakeLists specified `LLDB_PYTHON_RELATIVE_PATH` to be the path where Python modules are installed, relative to the install prefix (see https://github.com/llvm/llvm-project/blob/909d6c86eae32ef350ac35ba8564ed728544ac63/lldb/CMakeLists.txt#L41-L42). So I think it should be `lib/pythonX.Y`?

```
 lnguyen@xmake  /storage/python-test  ls -la lib/python3.8
total 4846
drwxr-xr-x  36 lnguyen  wheel     209 Feb 10 05:06 .
drwxr-xr-x   4 lnguyen  wheel       8 Feb 10 04:52 ..
-rw-r--r--   1 lnguyen  wheel   13919 Feb 10 05:06 LICENSE.txt
-rw-r--r--   1 lnguyen  wheel    5147 Feb 10 05:06 __future__.py
-rw-r--r--   1 lnguyen  wheel      64 Feb 10 05:06 __phello__.foo.py
drwxr-xr-x   2 lnguyen  wheel     518 Feb 10 05:07 __pycache__
-rw-r--r--   1 lnguyen  wheel    1801 Feb 10 05:06 _bootlocale.py
-rw-r--r--   1 lnguyen  wheel   26100 Feb 10 05:06 _collections_abc.py
-rw-r--r--   1 lnguyen  wheel    8749 Feb 10 05:06 _compat_pickle.py
-rw-r--r--   1 lnguyen  wheel    5340 Feb 10 05:06 _compression.py
-rw-r--r--   1 lnguyen  wheel    6027 Feb 10 05:06 _dummy_thread.py
-rw-r--r--   1 lnguyen  wheel   14598 Feb 10 05:06 _markupbase.py
-rw-r--r--   1 lnguyen  wheel   19600 Feb 10 05:06 _osx_support.py
-rw-r--r--   1 lnguyen  wheel    6189 Feb 10 05:06 _py_abc.py
-rw-r--r--   1 lnguyen  wheel  228666 Feb 10 05:06 _pydecimal.py
-rw-r--r--   1 lnguyen  wheel   93177 Feb 10 05:06 _pyio.py
-rw-r--r--   1 lnguyen  wheel    3115 Feb 10 05:06 _sitebuiltins.py
-rw-r--r--   1 lnguyen  wheel   25268 Feb 10 05:06 _strptime.py
-rw-r--r--   1 lnguyen  wheel   25994 Feb 10 05:06 _sysconfigdata__win_.py
-rw-r--r--   1 lnguyen  wheel    7220 Feb 10 05:06 _threading_local.py
-rw-r--r--   1 lnguyen  wheel    5735 Feb 10 05:06 _weakrefset.py
# many more lines omitted
```

Same issue with `LLDB_PYTHON_HOME`. In the CMakeLists, it's set to `PYTHON_HOME` by default, so separating the two might be problematic.

Maybe what you want is separating LLDB lib from Python stdlib, I'm gonna do some research on it.

Reply:
There's also the problem of relocatability. We need to be able to move the toolchain and still have LLDB find the correct `PYTHONHOME`. This can be done by setting `LLDB_PYTHON_HOME` to a path relative to where `liblldb` is, which would be `..`.